File: DataStructure/1.Stack/Group1/seoyoon/stack_4.cpp
#include <iostream>
#include <string>
#include <stack>
using namespace std;

bool isGoodWord(const string s);

int main()
{
	int num;
	int count = 0;
	string s;

	cin >> num;
	for (int i = 0; i < num; i++)
	{
		cin >> s;
		if (isGoodWord(s))
			count++;
	}
	cout << count;

	return 0;
}

bool isGoodWord(const string s)
{
	stack<char> stk;

	for (char ch : s)
	{
		if (!stk.empty() && ch == stk.top())
			stk.pop();
		else
			stk.push(ch);
	}

	return stk.empty();
}

File: DataStructure/1.Stack/Group1/yumin/stack4.py
n = int(input())
ans = 0

for i in range(n):
    stack = []
    for j in input():
        if len(stack) == 0:
            stack.append(j)
        else:
            if j == "A":
                if stack[-1] == "A":
                    stack.pop()
                elif stack[-1] == "B":
                    stack.append(j)
            elif j == "B":
                if stack[-1] == "B":
                    stack.pop()
                elif stack[-1] == "A":
                    stack.append(j)
    if len(stack) == 0:
        ans += 1

print(ans)